src/main/java/org/mifos/connector/channel/validator/ChannelValidator.java

```diff
@@ -0,0 +1,267 @@
+package org.mifos.connector.channel.validator;
+
+import static org.mifos.connector.common.exception.PaymentHubError.ExtValidationError;
+
+import org.mifos.connector.channel.utils.ChannelValidatorsEnum;
+import org.mifos.connector.common.channel.dto.PhErrorDTO;
+import org.mifos.connector.common.channel.dto.TransactionChannelRequestDTO;
+import org.mifos.connector.common.exception.PaymentHubErrorCategory;
+import org.mifos.connector.common.gsma.dto.GsmaTransfer;
+import org.mifos.connector.common.validation.ValidatorBuilder;
+
+@SuppressWarnings("HideUtilityClassConstructor")
+public class ChannelValidator {
+
+    private static final String resource = "ChannelValidator";
+    private static final String payer = "payer";
+    private static final String partyIdInfo = "partyIdInfo";
+    private static final String partyIdType = "partyIdType";
+    private static final String partyIdentifier = "partyIdentifier";
+    private static final String partyIdIdentifier = "partyIdIdentifier";
+    private static final String payee = "payee";
+    private static final String amount = "amount";
+    private static final String amount_value = "amount";
+    private static final String currency = "currency";
+    private static final int expectedCurrencyLength = 3;
+    private static final String requestingOrganisationTransactionReference = "requestingOrganisationTransactionReference";
+    private static final String subType = "subType";
+    private static final String type = "type";
+    private static final String descriptionText = "descriptionText";
+    private static final String requestDate = "requestDate";
+    private static final String customData = "customData";
+    private static final String key = "key";
+    private static final String value = "value";
+
+    public static PhErrorDTO validateTransfer(TransactionChannelRequestDTO request) {
+        final ValidatorBuilder validatorBuilder = new ValidatorBuilder();
+
+        // validating payer
```

Comment:
More generic validation methods have been added to the Validator Builder. Please check the bill-pay service validations for clarity and use them.

Reply:
ok